baseplate.go

```diff
@@ -325,21 +345,33 @@ func New(ctx context.Context, args NewArgs) (context.Context, Baseplate, error)
 	closer, err := log.InitSentry(cfg.Sentry)
 	if err != nil {
 		bp.Close()
-		return nil, nil, err
+		return nil, nil, fmt.Errorf(
+			"baseplate.New: failed to init sentry from config %#v: %w",
+			cfg.Sentry,
+			err,
+		)
```

Comment:
🔕 I usually print the error on the first line and then the rest on a subsequent line.  ala
```
fmt.Errorf("failed to init sentry from config: %s\n%s", err, marshalForError(cfg.Sentry))
```

(I also don't rely on the %#v output since it's pretty rough to parse as a reader, also 🔕 )

Reply:
same for below errors

Reply:
I consider `\n` an abnormal in error messages and should be avoided. Sure if you have a very very long error message maybe an `\n` is inevitable 🤷 , but it should be avoided as much as possible.

Both golint and staticcheck will treat capitalized error message or error messages ending with `.` as an error, which is inline with "don't put `\n` into your error messages".

Reply:
The first rule of error messages is that they should be useful.  Splatting a config into the error with %#v almost always violates this because it's horribly opaque, the user didn't write the config in Go syntax, it doesn't follow pointers, and it's all smashed into one unbroken blob.  So, it's "less bad" to stuff it in the second line.

The alternative is to log the bad config and then return an error saying to scroll up and look at the error, which isn't a great look, but is still serviceable.

Reply:
are you ok with `%+v`?

Reply:
+v doesn't really solve the problem though.  It's not Go syntax, sure, but it's still a condensed opaque syntax that doesn't match what the user typed in

Reply:
and I don't think +v follows pointers any more than #v, right?

Reply:
I would say dumping the config here (with either `%#v` or `%+v`, but `%v` is not good enough) is providing useful information. it will make it obvious to the user if the error is caused by the parsed config is actually empty (so they might have a wrong filename passed in, or maybe a typo in the yaml file, or something like that).

But I agree with you that it's less important than the error itself, so I swapped the order of them (by using `baseplate.New: ...: %w (config: %#v)` format string)